bubblegum/xrf/view/guessparam.enaml

```diff
@@ -64,199 +66,29 @@ enamldef GuessParamView(DockItem): param_view:
-                            else:
-                                fd7.value = param_m.param_d_perm['fwhm_offset']['value']
-                                fd7.enabled = False
-                                pb7.text = 'user value'
-
-                    Label: lbl8:
-                        text = 'fwhm Coef, b2'
-                        tool_tip = 'width**2 = (b1/2.3548)**2 + 3.85*b2*E, 3.85keV is electron-hole pair creation energy in silicon '
-                    FloatField: fd8:
-                        enabled = False
-                        value := param_m.param_d['fwhm_fanoprime']['value']
-                        maximum_size = 100
-                    PushButton: pb8:
-                        text = 'user value'
-                        checkable = True
-                        clicked ::
-                            if checked:
-                                fd8.enabled = True
-                                pb8.text = 'default value'
-                            else:
-                                fd8.value = param_m.param_d_perm['fwhm_fanoprime']['value']
-                                fd8.enabled = False
-                                pb8.text = 'user value'
-
-                    Label: lbl9:
-                        text = 'Compton fwhm Coef'
-                    FloatField: fd9:
-                        enabled = False
-                        value := param_m.param_d['compton_fwhm_corr']['value']
-                        maximum_size = 100
-                    PushButton: pb9:
-                        text = 'user value'
-                        checkable = True
-                        clicked ::
-                            if checked:
-                                fd9.enabled = True
-                                pb9.text = 'default value'
-                            else:
-                                fd9.value = param_m.param_d_perm['compton_fwhm_corr']['value']
-                                fd9.enabled = False
-                                pb9.text = 'user value'
-
+                Form:
+                    Looper: looper:
+                        iterable << param_model.parameters.values()
+                        Label:
+                            text = loop_item.description
+                            tool_tip = loop_item.tool_tip
+                        Container:
+                            padding = Box(0, 0, 0, 0)
+                            constraints = [hbox(ff, pb)]
+                            FloatField: ff:
+                                minimum := loop_item.min
+                                maximum := loop_item.max
+                                value := loop_item.value
+                            PushButton: pb:
+                                text = 'Default'
+                                clicked ::
+                                    ff.value = loop_item.default_value
+                    row_spacing = -1
```

Comment:
Would sorting in alphabetical order be ok? Or do you want to be able to specify exactly what order they appear in?